new FAQ information (#12607)
DataDog/documentation: content/en/agent/faq/log4j_mitigation.md

If you are using the Datadog Agent between versions v7.17.0/v6.17.0 and v7.32.2/v6.32.2, you may be impacted by the vulnerability presented by Log4Shell (CVE-2021-44228). If you are using an Agent earlier than v7.17.0/v6.17.0, you should not be impacted by the vulnerability unless you configured log4j to log with the JMS Appender (an option that is not supported by the Agent, but if you did it, disable the appender).

**If you are on an impacted version, to mitigate the vulnerability, the best option is to upgrade your Datadog Agent to v7.32.3 (v6.32.3) or later.**

> ## Seeing if your Agent Version is vulnerable

Reviewer: ```suggestion
## Seeing if your Agent version is vulnerable
```

### With a dashboard

> To see if your Datadog Agent (>= 6.17.0 - <= 6.32.2; >= 7.17.0 - <= 7.32.2) is the recommended version (6.32.3 and 7.32.3) and not running on a Log4j vulnerable version, [import][1] this dashboard template to your Datadog account:

Reviewer: ```suggestion
To see if your Datadog Agent (>= 6.17.0 - <= 6.32.2; >= 7.17.0 - <= 7.32.2) is the recommended version (6.32.3 and 7.32.3) and not running on a Log4j vulnerable version, [import][1] the following dashboard template to your Datadog account:
```

> [Agent version template][2]

Reviewer: ```suggestion
[**Datadog Agent Version Check dashboard template**][2]
```

</br>
</br>

> {{< img src="agent/faq/dashboard.png" alt="Integrations" >}}

Reviewer: ```suggestion
{{< img src="agent/faq/dashboard.png" alt="Datadog Agent Version Check dashboard showing vulnerable Agents" >}}
```

> If you want to use this dashboard over multiple Datadog accounts or hosts, use the API to automate the process. Run this command in the directory where the JSON file is saved:

Reviewer: ```suggestion
To create multiple version of this dashboard for multiple Datadog accounts or hosts, you can automate the creation process with the dashboards API. Run the following command in the directory where the JSON file is saved:
```

```curl
curl -X POST "https://api.datadoghq.com/api/v1/dashboard" \
-H "Content-Type: application/json" \
-H "DD-API-KEY: ${DD_API_KEY}" \
-H "DD-APPLICATION-KEY: ${DD_APP_KEY}" \
-d @DatadogAgentVersionCheck.json
```

> **Note**: This dashboard does not show older versions of the Datadog Agent (v5), because those versions are not vulnerable.

Reviewer: ```suggestion
**Note**: The Datadog Agent Version Check dashboard does not show older versions of the Datadog Agent (v5), because those versions are not vulnerable.
```

### With the CLI

You can also check specific Agent version information with the Agent CLI `version` subcommand. For more information, see the [Agent CLI documentation][3].

## Upgrading your Agent

To update the Datadog Agent core between two minor versions on your host or container, run the [install command for your platform][4].

## If you can't upgrade your Agent version

If you are not able to upgrade your Agent at this time, use these instructions to implement an environment variable (`LOG4J_FORMAT_MSG_NO_LOOKUPS="true"` on the JMXFetch process or the Agent process) to partially mitigate the vulnerability: 

> **Note**: If you are running v7.32.2 or v6.32.2, you do not need to perform these steps. The Agent v7.32.2 (and v6.32.2) [starts jmxfetch with a property][5] that achieves the same result. In all cases, the best option is to upgrade your Datadog Agent to v7.32.3 (v6.32.3) or later.

Reviewer: ```suggestion
**Note**: If you are running v7.32.2 or v6.32.2, you do not strictly need to perform these steps. The Agent v7.32.2 (and v6.32.2) [starts jmxfetch with a property][5] that achieves the same result. However, in all cases, the best option is to upgrade your Datadog Agent to v7.32.3 (v6.32.3) or later.
```

> **Note**: Setting the LOG4J_FORMAT_MSG_NO_LOOKUPS environment variable to true will reduce the risk of remote code execution but it is not a complete mitigation.

Reviewer: ```suggestion
**Note**: Setting the `LOG4J_FORMAT_MSG_NO_LOOKUPS` environment variable to `true` will reduce the risk of remote code execution but it is not a complete mitigation.
```

**The best way to mitigate the vulnerability is to upgrade your Datadog Agent to v7.32.3 (v6.32.3) or later.**

> If you are not sure which version of the Agent you are using, see [Seeing if your Agent Version is vulnerable](#seeing-if-your-agent-version-is-vulnerable).

Reviewer: ```suggestion
If you are not sure which version of the Agent you are using, read [Seeing if your Agent Version is vulnerable](#seeing-if-your-agent-version-is-vulnerable).
```
